src/main.cpp

```diff
@@ -9,6 +9,142 @@
 
 namespace fs = filesystem;
 
+void ArgConfig( args::ArgumentParser &parser) {
+
+    Config config;
```

Comment:
This is wrong, take `config` as a reference from the `config` declared in `main()`.